False positive for `logging-fstring-interpolation`

### Bug description

When using f-string without arguments in a logging message, Pylint would warn me about not using lazy % formatting, even though I do use it.

a.py:
```python
import logging
logging.error(f'Hello %s', 'World!')
```

output:
`ERROR:root:Hello World!`


### Configuration

_No response_

### Command used

```shell
pylint a.py
```


### Pylint output

```shell
************* Module a
a.py:2:0: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
a.py:2:14: W1309: Using an f-string that does not have any interpolated variables (f-string-without-interpolation)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```


### Expected behavior

Report only `f-string-without-interpolation`

### Pylint version

```shell
pylint 2.10.2
astroid 2.7.3
Python 3.9.4 (default, Apr 30 2021, 18:51:54) 
[GCC 10.2.0]
```


### OS / Environment

Manjaro Linux, Kernel 5.10

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/4984.false_positive
+++ b/doc/whatsnew/fragments/4984.false_positive
@@ -1,0 +1,3 @@
+Fix ``logging-fstring-interpolation`` false positive raised when logging and f-string with ``%s`` formatting.
+
+Closes #4984

--- a/pylint/checkers/logging.py
+++ b/pylint/checkers/logging.py
@@ -105,12 +105,7 @@ CHECKED_CONVENIENCE_FUNCTIONS = {
     "warning",
 }
 
-MOST_COMMON_FORMATTING = {
-    "%s",
-    "%d",
-    "%f",
-    "%r",
-}
+MOST_COMMON_FORMATTING = frozenset("%s", "%d", "%f", "%r")
 
 
 def is_method_call(

--- a/pylint/checkers/logging.py
+++ b/pylint/checkers/logging.py
@@ -403,7 +403,7 @@ def _count_supplied_tokens(args: list[nodes.NodeNG]) -> int:
     return sum(1 for arg in args if not isinstance(arg, nodes.Keyword))
 
 
-def str_formatting_in_f_string(node: nodes.NodeNG) -> bool:
+def str_formatting_in_f_string(node: nodes.JoinedStr) -> bool:
     """Determine whether the node represents an f-string with string formatting.
 
     For example: `f'Hello %s'`
@@ -416,6 +416,5 @@ def str_formatting_in_f_string(node: nodes.NodeNG) -> bool:
     )
 
 
-# Detect % formatting inside an f-string, such as `f'Hello %s'`
 def register(linter: PyLinter) -> None:
     linter.register_checker(LoggingChecker(linter))

--- a/pylint/checkers/logging.py
+++ b/pylint/checkers/logging.py
@@ -105,6 +105,13 @@ CHECKED_CONVENIENCE_FUNCTIONS = {
     "warning",
 }
 
+MOST_COMMON_FORMATTING = {
+    "%s",
+    "%d",
+    "%f",
+    "%r",
+}
+
 
 def is_method_call(
     func: bases.BoundMethod, types: tuple[str, ...] = (), methods: tuple[str, ...] = ()
@@ -262,12 +269,7 @@ class LoggingChecker(checkers.BaseChecker):
         elif isinstance(format_arg, nodes.Const):
             self._check_format_string(node, format_pos)
         elif isinstance(format_arg, nodes.JoinedStr):
-            if any(
-                "%s" in val.value
-                for val in format_arg.values
-                if isinstance(val, nodes.Const)
-            ):
-                # Detect % formatting inside an f-string, such as `f'Hello %s'`
+            if str_formatting_in_f_string(format_arg):
                 return
             self.add_message(
                 "logging-fstring-interpolation",
@@ -401,5 +403,19 @@ def _count_supplied_tokens(args: list[nodes.NodeNG]) -> int:
     return sum(1 for arg in args if not isinstance(arg, nodes.Keyword))
 
 
+def str_formatting_in_f_string(node: nodes.NodeNG) -> bool:
+    """Determine whether the node represents an f-string with string formatting.
+
+    For example: `f'Hello %s'`
+    """
+    # Check "%" presence first for performance.
+    return any(
+        "%" in val.value and any(x in val.value for x in MOST_COMMON_FORMATTING)
+        for val in node.values
+        if isinstance(val, nodes.Const)
+    )
+
+
+# Detect % formatting inside an f-string, such as `f'Hello %s'`
 def register(linter: PyLinter) -> None:
     linter.register_checker(LoggingChecker(linter))

--- a/pylint/checkers/logging.py
+++ b/pylint/checkers/logging.py
@@ -240,8 +240,9 @@ class LoggingChecker(checkers.BaseChecker):
         else:
             return
 
-        if isinstance(node.args[format_pos], nodes.BinOp):
-            binop = node.args[format_pos]
+        format_arg = node.args[format_pos]
+        if isinstance(format_arg, nodes.BinOp):
+            binop = format_arg
             emit = binop.op == "%"
             if binop.op == "+":
                 total_number_of_strings = sum(
@@ -256,11 +257,18 @@ class LoggingChecker(checkers.BaseChecker):
                     node=node,
                     args=(self._helper_string(node),),
                 )
-        elif isinstance(node.args[format_pos], nodes.Call):
-            self._check_call_func(node.args[format_pos])
-        elif isinstance(node.args[format_pos], nodes.Const):
+        elif isinstance(format_arg, nodes.Call):
+            self._check_call_func(format_arg)
+        elif isinstance(format_arg, nodes.Const):
             self._check_format_string(node, format_pos)
-        elif isinstance(node.args[format_pos], nodes.JoinedStr):
+        elif isinstance(format_arg, nodes.JoinedStr):
+            if any(
+                "%s" in val.value
+                for val in format_arg.values
+                if isinstance(val, nodes.Const)
+            ):
+                # Detect % formatting inside an f-string, such as `f'Hello %s'`
+                return
             self.add_message(
                 "logging-fstring-interpolation",
                 node=node,

--- a/pylint/checkers/logging.py
+++ b/pylint/checkers/logging.py
@@ -105,7 +105,7 @@ CHECKED_CONVENIENCE_FUNCTIONS = {
     "warning",
 }
 
-MOST_COMMON_FORMATTING = frozenset("%s", "%d", "%f", "%r")
+MOST_COMMON_FORMATTING = frozenset(["%s", "%d", "%f", "%r"])
 
 
 def is_method_call(

--- a/tests/functional/l/logging/logging_fstring_interpolation_py37.py
+++ b/tests/functional/l/logging/logging_fstring_interpolation_py37.py
@@ -8,3 +8,4 @@ WORLD = "world"
 logging.error(f'Hello {WORLD}')  # [logging-fstring-interpolation]
 
 logging.error(f'Hello %s', 'World!')  # [f-string-without-interpolation]
+logging.error(f'Hello %d', 1)  # [f-string-without-interpolation]

--- a/tests/functional/l/logging/logging_fstring_interpolation_py37.py
+++ b/tests/functional/l/logging/logging_fstring_interpolation_py37.py
@@ -3,3 +3,8 @@ import logging
 
 VAR = "string"
 logging.error(f"{VAR}")  # [logging-fstring-interpolation]
+
+WORLD = "world"
+logging.error(f'Hello {WORLD}')  # [logging-fstring-interpolation]
+
+logging.error(f'Hello %s', 'World!')  # [f-string-without-interpolation]

--- a/tests/functional/l/logging/logging_fstring_interpolation_py37.txt
+++ b/tests/functional/l/logging/logging_fstring_interpolation_py37.txt
@@ -1,3 +1,4 @@
 logging-fstring-interpolation:5:0:5:23::Use lazy % formatting in logging functions:UNDEFINED
 logging-fstring-interpolation:8:0:8:31::Use lazy % formatting in logging functions:UNDEFINED
 f-string-without-interpolation:10:14:10:25::Using an f-string that does not have any interpolated variables:UNDEFINED
+f-string-without-interpolation:11:14:11:25::Using an f-string that does not have any interpolated variables:UNDEFINED

--- a/tests/functional/l/logging/logging_fstring_interpolation_py37.txt
+++ b/tests/functional/l/logging/logging_fstring_interpolation_py37.txt
@@ -1,1 +1,3 @@
 logging-fstring-interpolation:5:0:5:23::Use lazy % formatting in logging functions:UNDEFINED
+logging-fstring-interpolation:8:0:8:31::Use lazy % formatting in logging functions:UNDEFINED
+f-string-without-interpolation:10:14:10:25::Using an f-string that does not have any interpolated variables:UNDEFINED

